### DISCUSSION TITLE
Latency on lease revoke/expriation

### DISCUSSION THREAD

**Discussion by ghamilton61** [etcd-io/etcd#14527] [General]
I'm trying to get a feel for what to expect in terms of the delay between the time a lease expires or is revoked and when the transaction to remove the lease and associated kvs takes place.  I've read some about 'expiredLeaseRetryInterval' and the process of sending expired leases to "an expired channel for revoking".  I've read that the time between expiration and deletion is generally around 0.5 seconds, but that under a heavy load it may take as much as 5 - 8 seconds.  But this was from one source and I've not been able to find other articles where this is discussed.  Can anyone provide me with benchmarks on this or expected latency?

**Comment by SimFG** [comment]
I find a benchmark example about the lease, detail: https://github.com/etcd-io/etcd/blob/main/tools/benchmark/cmd/lease.go, but there isn't a case that you need. Maybe you can design some tests for your need referring to `lease.go`
